### ISSUE TITLE
When csrftoken is set as HttpOnly-Cookie, X-CSRFTOKEN Header is "" for Ajax-Requests

### ISSUE THREAD

**Issue by jakicoll** [modoboa/modoboa-radicale#38]
When I try to create a new calender as normal user, after setting the name and color and clicking Submit, the windows does not disappear and no calendar is created. So I investigated this a little:

Chromium reports that the csrftoken cookie is set as HttpOnly-Cookie. Therefor it can not be accessed by javascript (document.cookies just returns an empty string), and thus no valid X-CSRFTOKEN Header is sent.

Creating new Folders in Webmail works well, the csrf-token is sent with the post data in this case. I guess something similar needs to be done here?

I'm on modoboa 1.10.6, modoboa-radicale 1.3.6 running in a python3 virtual environment with dependencies from pip.

Request:
> POST /api/v1/user-calendars/ HTTP/1.1
> Host: <redacted>
> User-Agent: Mozilla/5.0 (X11; Linux x86_64; rv:52.0) Gecko/20100101 Firefox/52.0
> Accept: application/json, text/plain, */*
> Accept-Language: en-US,en;q=0.5
> Accept-Encoding: gzip, deflate, br
> X-CSRFToken: 
> Content-Type: application/json;charset=utf-8
> x-requested-with: XMLHttpRequest
> Referer: <redacted>
> Content-Length: 35
> Cookie: csrftoken=*****************************; sessionid=******************************
> Connection: keep-alive
 
Post-Data
> {"color":"#B0BC00","name":"dfgdfg"}
>
Reponse is 403
> {detail: "CSRF Failed: CSRF token missing or incorrect."}

**Comment by tonioo** [comment]
@jakicoll Your trace says you're using FF, not chromium... Can you tell me which versions please?

**Comment by jakicoll** [comment]
I guess it's because you webserver does not mark the cookie as HttpOnly.

I think that somewhere in Django a default setting has changed and the cookie needs to be marked as JavaScript readable explicitly from now on.

I hope I can look into in this week and will tell you if I found a cause. Unfortunately I'm not really familiar with Django yet, but time will come... :)

Am 25. September 2018 13:26:03 MESZ schrieb Antoine Nguyen <notifications@github.com>:
>I'm using Debian stable too (stretch) and I'm able to create calendars
>on both FF and chromium. Versions I use: FF 60.2.0esr and Chromium
>69.0.3497.92.
>
>-- 
>You are receiving this because you were mentioned.
>Reply to this email directly or view it on GitHub:
>https://github.com/modoboa/modoboa-radicale/issues/38#issuecomment-424305768

-- 
Von meinem Fairphone mit K-9 Mail gesendet.

**Comment by tonioo** [comment]
Can you check if your settings.py contains the following setting: 
CSRF_COOKIE_HTTPONLY ?

Le 25/09/2018 à 13:30, Jakob S. a écrit :

**Comment by jakicoll** [comment]
Okay, it was entirely my fault; my webserver flags all outgoing cookies as httponly. :blush:

I didn't look at this first, because my browser had other cookies from services running on the same vhost, that didn't have a HttpOnly flag set. (So I concluded, the flag was set by the modoa/django). Turns out, those cookies were written with javascript on client side...

So, if anyone else ever get's stuck at this: Investiate your webserver configuration closely, it ain't a modoboa / django bug (unless you really did set ```CSRF_COOKIE_HTTPONLY``` explicitly to true - it defaults to false).

Thanks again @tonioo for you support and all the effort you put into modoboa. :+1:
